apps/tvremote/app.js

```diff
@@ -0,0 +1,655 @@
+  }
+};
+
+
+function swipeHandler(LR, UD) {
+  if (LR == -1) { // swipe left
+    if (currentScreen === "power") {
+      assignScreen("apps");
+      appMenu();
+
+    } else if (currentScreen === "apps") {
+      //require("widget_utils").hide();
+      assignScreen("selection");
+      E.showScroller();
+      prepareScreen(selection, selectionGrid, COLORS.DEFAULT, KEY_AREA);
+      drawKeys(KEY_AREA, selection);
+
+    } else if (currentScreen === "volume") {
+      sendPost("fast_forward");
+
+    } else if (currentScreen === "selection") {
+      sendPost("enter");
+    }
+  }
+  if (LR == 1) { // swipe right
+    if (currentScreen === "apps") {
+      assignScreen("power");
+      E.showScroller();
+      powerScreen();
+    } else if (currentScreen === "volume") {
+      sendPost("rewind");
+    } else if (currentScreen === "selection") {
+      sendPost("back");
+    }
+  }
+  if (UD == -1) { // swipe up
+    if (currentScreen === "selection") {
+      assignScreen("volume");
+      prepareScreen(volume, volumeGrid, COLORS.DEFAULT, KEY_AREA);
+      drawKeys(KEY_AREA, volume);
+    } else if (currentScreen === "volume") {
+      sendPost("enter");
+    } else if (currentScreen === "ip") {
+      buttonPress(".", "ip");
+    } else if (currentScreen == "numbers") {
+      assignScreen("selection");
+      prepareScreen(selection, selectionGrid, COLORS.DEFAULT, KEY_AREA);
+      drawKeys(KEY_AREA, selection);
+    }
+  }
+  if (UD == 1) { // swipe down
+    if (currentScreen === "volume") {
+      assignScreen("selection");
+      prepareScreen(selection, selectionGrid, COLORS.DEFAULT, KEY_AREA);
+      drawKeys(KEY_AREA, selection);
+    } else if (currentScreen === "selection") {
+      assignScreen("numbers");
+      prepareScreen(numbers, numbersGrid, COLORS.DEFAULT, KEY_AREA);
+      drawKeys(KEY_AREA, numbers);
+    }
+  }
+}
+Bangle.on('swipe', swipeHandler);
+
+
+function touchHandler(button, e) {
+  const screenActions = {
+    ip: () => checkButtons(numbers),
+    volume: () => checkButtons(volume),
+    numbers: () => checkButtons(numbers),
+    selection: () => checkButtons(selection),
+    power: () => {
+      if (Math.pow(e.x - 88, 2) + Math.pow(e.y - 88, 2) < 2500) {
+        sendPost("power");
+      }
+    }
+  };
+
+  function checkButtons(buttonMap) {
+    for (let key in buttonMap) {
+      if (typeof buttonMap[key] === "undefined") continue;
+      let r = buttonMap[key].xy;
+
+      if (e.x >= r[0] && e.y >= r[1] && e.x < r[2] && e.y < r[3]) {
+        if (currentScreen === "ip" && key === "ok") {
+          subMenu();
+        } else {
+          buttonPress("" + key, currentScreen);
+        }
+      }
+    }
+  }
+
+  if (currentScreen in screenActions) screenActions[currentScreen]();
+}
+Bangle.on('touch', touchHandler);
+
+Bangle.loadWidgets();
+Bangle.drawWidgets();
+
+if (serverData === undefined) {
+  E.showAlert(`${serverDataFile}.json missing.\nSee READ.me`, "Config Error").then(function() {
```

Comment:
![download](https://github.com/user-attachments/assets/16796810-b98e-4af9-bb14-95e0a2e7b6ca)

The alert body gets a little long. I suggest:


```suggestion
  E.showAlert(`No settings.\nSee READ.me`, "Config Error").then(function() {
```

Reply:
happy with the amendment. I wasn't sure how detailed to be with the alert. But if I can get the customiser working, this will be sutable for the interim